api-wrapper/services/api-wrapper/src/main/java/net/catenax/edc/apiwrapper/connector/sdk/service/ContractOfferService.java

```diff
@@ -1,64 +1,79 @@
 package net.catenax.edc.apiwrapper.connector.sdk.service;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
-import jakarta.ws.rs.BadRequestException;
 import jakarta.ws.rs.InternalServerErrorException;
+import net.catenax.edc.apiwrapper.config.ApiWrapperConfig;
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
 import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
 import org.eclipse.dataspaceconnector.spi.types.TypeManager;
 import org.eclipse.dataspaceconnector.spi.types.domain.catalog.Catalog;
 import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractOffer;
+import org.jetbrains.annotations.NotNull;
 
 import java.io.IOException;
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
 
 import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.groupingBy;
 
 public class ContractOfferService {
     private final Monitor monitor;
     private final ObjectMapper objectMapper;
     private final OkHttpClient httpClient;
+    private final ApiWrapperConfig config;
+    private final Map<String, Map<String, List<ContractOffer>>> byAssetIdCache = new ConcurrentHashMap<>();
```

Comment:
I'm not sure, but I think the name of the variable is somehow a little bit misleading. For me it sounds like, that the assetId would be the key (the root key). What do you think about adding a short comment here?

```suggestion
    // ProviderURL -> AssetId -> List<ContractOffer>
    private final Map<String, Map<String, List<ContractOffer>>> byAssetIdCache = new ConcurrentHashMap<>();
```